tests/components/blebox/conftest.py

```diff
@@ -59,47 +65,96 @@ def mock_feature(category, spec, **kwargs):
 
 def mock_config(ip_address="172.100.123.4"):
     """Return a Mock of the HA entity config."""
-    return MockConfigEntry(
-        domain=const.DOMAIN, data={CONF_HOST: ip_address, CONF_PORT: 80},
-    )
+    return MockConfigEntry(domain=DOMAIN, data={CONF_HOST: ip_address, CONF_PORT: 80})
+
+
+@pytest.fixture(name="config")
+def config_fixture():
+    """Create hass config fixture."""
+    return {DOMAIN: {CONF_HOST: "172.100.123.4", CONF_PORT: 80}}
+
 
+@pytest.fixture
+def wrapper(request):
+    """Return an entity wrapper from given fixture name."""
+    return request.getfixturevalue(request.param)
 
-class BleBoxTestHelper:
-    """Helper methods for tests."""
 
-    HASS_TYPE = None  # override in subclass
+class Wrapper:
```

Comment:
It's ok to use a helper for the set up part of the tests but not for the action or assertion. Please at most just make a helper function that sets up the integration.

Reply:
I'm more a fan of clearer intentions in tests (instead of removing the fascade, revealing details with the cost of more heavily relying on the implementation)...

... so just to make sure, is this what you expect:

```python
@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
async def test_open(wrapper, hass, config):
    """Test cover opening."""

    feature_mock = wrapper.feature_mock
    def initial_update():
        feature_mock.state = 3  # manually stopped

    def open_gate():
        feature_mock.state = 1  # opening

    feature_mock.async_update = AsyncMock(side_effect=initial_update)
    feature_mock.async_open = AsyncMock(side_effect=open_gate)

    entity_id = (await wrapper.setup(hass, config)).entity_id
    assert hass.states.get(entity_id).state == STATE_CLOSED

    feature_mock.async_update = AsyncMock(side_effect=None)
    await hass.services.async_call(
        "cover",
        SERVICE_OPEN_COVER,
        {"entity_id": entity_id},
        blocking=True,
    )
    assert hass.states.get(entity_id).state == STATE_OPENING

```

?

Reply:
I'm assuming that `feature_mock` is part of the library. And `async_setup_entity` sets up the integration. I think it looks good.